Git Formats/Git Log.sublime-syntax

```diff
@@ -7,67 +7,65 @@ version: 2
+first_line_match: '{{commit_begin}}'
 
 contexts:
-  prototype:
-    - include: Git Common.sublime-syntax#comments-line
-
   main:
+    - include: Git Common.sublime-syntax#comments-line
     # 1st header line
     # commit d9d9fb804f5d61c13ba2f8746af33a9f3c609075
     - match: '{{commit_begin}}'
       captures:
         1: keyword.other.commit.git.log
         2: constant.other.hash.git.log
-      push:
-        - meta_scope: meta.header.git.commit markup.raw.block.git.log
-        - match: $\n?
-          scope: meta.header.git.commit markup.raw.block.git.log
-          pop: 1
-          embed: commit-header
-          escape: '{{commit_or_diff_begin}}'
-        - match: (\()(HEAD)\s*(->)
-          captures:
-            1: punctuation.section.parens.begin.git.log
-            2: support.type.git.log
-            3: punctuation.separator.mapping.git.log
-          push:
-            - match: \)
-              scope: punctuation.section.parens.end.git.log
-              pop: 1
-            - match: \b(?:master|main)\b
-              scope: entity.other.branch-name.master.git.log
-            - match: (\w+)(?:(/)(\w+))?
-              scope: entity.other.branch-name.git.log
-            - match: ','
-              scope: punctuation.separator.sequence.git.log
+      embed: commit-decoration
+      escape: '{{commit_or_diff_begin}}'
+    # diff output
     - match: (?={{diff_begin}})
       embed: Git Diff.sublime-syntax#diffs
       embed_scope: source.diff.git
       escape: (?={{commit_begin}})
 
-  commit-header:
-    # All header attributes are mappings of `key: value` format.
-    # For simplicity we do not want to distinguish the entries.
-    # Author: first lastname <mail@domain.com>
-    # Date:   Wed Sep 13 22:42:14 2017 +0200
-    - match: ^(\w+)\s*(:)\s*
+  commit-decoration:
+    - meta_scope: meta.header.git.commit
+    - match: \(
+      scope: punctuation.section.parens.begin.git.log
+      push: commit-decoration-body
+    - match: ^
+      set: commit-header
+
+  commit-decoration-body:
+    - meta_scope: meta.parens.decoration.log
+    - match: \)
+      scope: punctuation.section.parens.end.git.log
+      pop: 1
+    - match: (HEAD)\s*(->)
+      captures:
+        1: support.type.git.log
+        2: punctuation.separator.mapping.git.log
+    - match: \b(?:master|main)\b
+      scope: entity.other.branch-name.master.git.log
+    - match: \b(tag)(:)\s*([\w./-]+)
       captures:
-        1: keyword.other.header.git.log
-        2: punctuation.separator.key-value.git.log
-      push:
-        - meta_scope: meta.header.git.commit markup.raw.block.git.log
-        - meta_content_scope: string.unquoted.log
-        - match: \n
-          pop: 1
-        - include: Git Common.sublime-syntax#email
-    # using push instead of include as workaround for
-    # https://github.com/SublimeTextIssues/Core/issues/2395
+        2: punctuation.separator.log
+        3: entity.other.tag.git.log
```

Comment:
> Any opinion on scope for the word "tag"?

This one?

`entity.other.tag` was just used as branch names are also scoped `entity.other`.

It's actually not a mapping, hence the `->` above may also just be simple separator.

In general I have no strong opinion about best choise for scopes as all git related syntaxes are such special purpose, they don't fit into common language's scope schemes.

I guess that's the reason for `entity` instead of `variable` vs. `constant`. But well, the git commit hash is also scoped `constant`. 

We should probably only try to use same sort of scopes in all git syntaxes.

Reply:
I meant this part with the part you quoted...

```
  tag: 0.1.0
  ^^^
```

But I am also interested in the tag itself, so no worries.